pymatgen/io/lammps/inputs.py

```diff
@@ -17,15 +17,439 @@
+            to the stage "Definition of the potential", simply use
+            your_input_file.add_command(
+                stage_name="Definition of the potential",
+                command="pair_coeff 1 1 morse 0.0580 3.987 3.404"
+            )
+            or
+            your_input_file.add_command(
+                stage_name="Definition of the potential",
+                command="pair_coeff",
+                args="1 1 morse 0.0580 3.987 3.404"
+            )
+
+        Args:
+            stage_name (str): name of the stage to which the command should be added.
+            command (str): LAMMPS command, with or without the arguments.
+            args (str): Arguments for the LAMMPS command.
+        """
+        if args is None:
+            string_split = command.split()
+            command = string_split[0]
+            args = " ".join(string_split[1:])
+
+        # Find where the command should be added (stage index instead of stage name)
+        idx = self.stages_names.index(stage_name)
+        # Add the command
+        if not self.input_settings[idx][1]:
+            self.input_settings[idx][1] = [[command, args]]
+        else:
+            self.input_settings[idx][1].append([command, args])
+
+    def add_comment(self, comment: str, inline: bool = False, stage_name: str = None, index_comment: bool = False):
+        """
+         Method to add a comment in a stage or as a whole stage (which will do nothing when lammps runs).
+
+        Args:
+            comment: Comment string to be added. The comment will be
+                preceded by '#' in the generated input.
+            inline: True if the comment should be inline within a given block of commands.
+            stage_name: set the stage_name to which the comment will be written. Required if inline is True.
+            index_comment: True if the comment should start with "Comment x" with x its number in the ordering.
+                Used only for inline comments.
+        """
+        self.ncomments += 1
+        # "Stage" of comment only
+        if not inline:
+            if stage_name is None:
+                stage_name = f"Comment {self.ncomments}"
+            self.add_stage(command="# " + comment, stage_name=stage_name)
+            self.nstages += -1
+        # Inline comment
+        elif inline and stage_name:
+            command = "#"
+            if index_comment:
+                if "Comment" in comment and comment.strip()[9] == ":":
+                    args = ":".join(comment.split(":")[1:])
+                else:
+                    args = comment
+            else:
+                args = comment
+            self.add_command(command=command, args=args, stage_name=stage_name)
+
+        else:
+            raise NotImplementedError("If you want to add an inline comment, please specify the stage name.")
+
+    def get_string(self, ignore_comments: bool = False, keep_stages: bool = False) -> str:
```

Comment:
I wonder if this will or can be written into a file?

Reply:
Yes, through the write_file method (see InputFile object in pymatgen.io.core, or the write_file method that I have now added here to override the InputFile method, because the signatures differ). This is used for instance by InputSets to write the input files.

Reply:
Super! Thanks for letting me know.